include/ginkgo/core/base/mpi.hpp

```diff
@@ -454,6 +454,57 @@ class communicator {
         this->comm_.reset(new MPI_Comm(comm_out), comm_deleter{});
     }
 
+    /**
+     * Creates a new communicator and takes ownership of the MPI_Comm.
+     *
+     * The ownership is shared with all communicators that are created as a
+     * copy from the new communicator.
+     * The underlying MPI_Comm will be freed when the last communicator with
+     * ownership is destroyed.
```

Comment:
does it fit the code?
the new communicator does not share the ownership with the input comm, right?

Reply:
I've clarified the documentation a bit.